ui/components/app/nfts-tab/nfts-tab.js

```diff
@@ -66,6 +71,33 @@ export default function NftsTab() {
   const showNftBanner = hasAnyNfts === false;
   const currentNetwork = useSelector(getCurrentNetwork);
   const currentLocale = useSelector(getCurrentLocale);
+  const hasAnyPreviouslyOwnedNfts = previouslyOwnedCollection.nfts.length > 0;
+  useEffect(() => {
+    if (
+      (isIpfsEnabled || openSeaEnabled) &&
+      (hasAnyNfts || hasAnyPreviouslyOwnedNfts)
+    ) {
+      const allNfts = previouslyOwnedCollection.nfts.concat(
+        Object.values(collections).flatMap((collection) => collection.nfts),
+      );
+      const nftsToUpdateMetadata = allNfts.filter(
+        (singleNft) =>
+          !singleNft.name && !singleNft.description && !singleNft.image,
+      );
+
+      if (nftsToUpdateMetadata.length !== 0) {
+        dispatch(updateNftMetadata(nftsToUpdateMetadata));
```

Comment:
Could the update be triggered just once when the ipfs/media settings are flipped on?  Instead of checking each time this renders?  

Worried there could be NFTs that actually have no `name`, `description`, `image` but this would always be trying to update them.